source/slang/hlsl.meta.slang

```diff
@@ -6936,6 +6936,57 @@ uint countbits(uint value)
     }
 }
 
+[__readNone]
+[require(cpp_cuda_glsl_hlsl_metal_spirv, shader5_sm_5_0)]
+uint2 countbits(uint2 value)
+{
+    __target_switch
+    {
+    case hlsl:
+        __intrinsic_asm "countbits";
+    case glsl:
+        __intrinsic_asm "bitCount";
+    case metal:
+        __intrinsic_asm "popcount";
+    default:
+        return uint2(countbits(value.x), countbits(value.y));
+    }
+}
+
+[__readNone]
+[require(cpp_cuda_glsl_hlsl_metal_spirv, shader5_sm_5_0)]
+uint3 countbits(uint3 value)
+{
+    __target_switch
+    {
+    case hlsl:
+        __intrinsic_asm "countbits";
+    case glsl:
+        __intrinsic_asm "bitCount";
+    case metal:
+        __intrinsic_asm "popcount";
+    default:
+        return uint3(countbits(value.x), countbits(value.y), countbits(value.z));
+    }
+}
+
+[__readNone]
+[require(cpp_cuda_glsl_hlsl_metal_spirv, shader5_sm_5_0)]
+uint4 countbits(uint4 value)
+{
+    __target_switch
+    {
+    case hlsl:
+        __intrinsic_asm "countbits";
+    case glsl:
+        __intrinsic_asm "bitCount";
+    case metal:
+        __intrinsic_asm "popcount";
+    default:
+        return uint4(countbits(value.x), countbits(value.y), countbits(value.z), countbits(value.w));
+    }
+}
```

Comment:
These 3 functions can be simplified by using:
`__generic<let N : int>` so the input can be a `vector<uint, N>`. The `default:` case can then be a `VECTOR_MAP_UNARY`